crates/task-impls/src/quorum_proposal.rs

```diff
@@ -80,8 +71,160 @@ struct ProposalDependencyHandle<TYPES: NodeType> {
     sender: Sender<Arc<HotShotEvent<TYPES>>>,
 
     /// Reference to consensus. The replica will require a write lock on this.
-    #[allow(dead_code)]
     consensus: Arc<RwLock<Consensus<TYPES>>>,
+
+    /// Output events to application
+    #[allow(dead_code)]
+    output_event_stream: async_broadcast::Sender<Event<TYPES>>,
+
+    /// Membership for Timeout votes/certs
+    #[allow(dead_code)]
+    timeout_membership: Arc<TYPES::Membership>,
+
+    /// Membership for Quorum Certs/votes
+    quorum_membership: Arc<TYPES::Membership>,
+
+    /// Our public key
+    public_key: TYPES::SignatureKey,
+
+    /// Our Private Key
+    private_key: <TYPES::SignatureKey as SignatureKey>::PrivateKey,
+
+    /// View timeout from config.
+    #[allow(dead_code)]
+    timeout: u64,
+
+    /// Round start delay from config, in milliseconds.
+    round_start_delay: u64,
+
+    /// The node's id
+    #[allow(dead_code)]
+    id: u64,
+}
+
+impl<TYPES: NodeType> ProposalDependencyHandle<TYPES> {
+    /// Sends a proposal if possible from the high qc we have
+    #[allow(clippy::too_many_lines)]
+    pub async fn publish_proposal_if_able(
+        &self,
+        view: TYPES::Time,
+        event_stream: &Sender<Arc<HotShotEvent<TYPES>>>,
+        commit_and_metadata: CommitmentAndMetadata<TYPES::BlockPayload>,
+    ) -> bool {
```

Comment:
If we're making any changes while copying over consensus logic, I think it would be really worthwhile to make this return `anyhow::Result<()>`.

But if we're just trying to copy the logic as-is, it's probably better to change it in a separate PR

Reply:
Marking this for next PR. I definitely plan to re-add the anyhow logic from my last change. I was hoping to avoid making this PR more of a headache to review than it already is.